docs/rfcs/0001-integration-with-dune.md

```diff
@@ -0,0 +1,169 @@
+- Start Date: 2021-05-31
+- RFC PR: https://github.com/melange-re/melange/pull/200
+
+# Summary
+
+To allow Melange users to leverage Dune `library` and `executable` stanzas as
+it is currently possible with any other OCaml project.
+
+# Basic example
+
+Dune would allow to use Melange as another cross-compiler, e.g.
+
+```dune
+(context (default (targets (melange))))
+```
+
+Then, users would be able to define `library` and `executable` stanzas that
+would be built by calling Melange binary `ocamlc`. To define the output format
+and extension, a new stanza `melange` would be supported in `dune` files:
+```dune
+(executable
+    (name foo)
+    (modes melange)
```

Comment:
This could also be inferred from the context, if a context has only `ocamlc` and it's a melange "ocamlc" all executables in this context must have `(modes melange)`

Reply:
> and it's a melange "ocamlc" 

how can Dune know this from just context? I thought from Dune perspective it's "just another" OCaml compiler installation.